src/fat_pointer.rs

```diff
@@ -31,4 +32,40 @@ impl FatPointer {
     pub fn into_u256(self) -> U256 {
         U256::zero() + unsafe { std::mem::transmute::<FatPointer, u128>(self) }
     }
+
+    pub fn validate(&self, is_fresh: bool) -> FatPointerValidationException {
```

Comment:
This shouldn't be needed. We already do this when constructing a fat pointer. (Otherwise the fuzzer would find crashes because bad fat pointers are dangerous.)